src/api/component.js

```diff
@@ -89,6 +89,8 @@ export default class Component extends HTMLElement {
       created(this);
     }
 
+    this.setAttribute('defined', '');
```

Comment:
It's probably best to add this before the `readyCallbacks` are fired so that the element has layout when those are called but this is up for discussion.